contracts/staking/test/utils/api_wrapper.ts

```diff
@@ -0,0 +1,254 @@
+import { ERC20ProxyContract } from '@0x/contracts-asset-proxy';
+import { artifacts as erc20Artifacts, DummyERC20TokenContract } from '@0x/contracts-erc20';
+import { BlockchainTestsEnvironment, constants as testUtilsConstants, txDefaults } from '@0x/contracts-test-utils';
+import { SignatureType } from '@0x/types';
+import { BigNumber, logUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { ContractAbi, ContractArtifact, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import {
+    artifacts,
+    EthVaultContract,
+    StakingContract,
+    StakingPoolRewardVaultContract,
+    StakingProxyContract,
+    ZrxVaultContract,
+} from '../../src';
+
+import { ApprovalFactory } from './approval_factory';
+import { constants } from './constants';
+import { SignedStakingPoolApproval, StakeInfo } from './types';
+
+export class StakingApiWrapper {
+    public stakingProxyContract: StakingContract; // Wrapped as StakingContract to borrow API
+    public stakingContract: StakingContract; // The real Staking.sol contract
+    public zrxVaultContract: ZrxVaultContract;
+    public ethVaultContract: EthVaultContract;
+    public rewardVaultContract: StakingPoolRewardVaultContract;
+    public zrxTokenContract: DummyERC20TokenContract;
+    public utils = {
+        // Epoch Utils
+        goToNextEpochAsync: async (): Promise<TransactionReceiptWithDecodedLogs> => {
+            const txReceipt = await this.stakingProxyContract.finalizeFees.awaitTransactionSuccessAsync();
+            logUtils.log(`Finalization costed ${txReceipt.gasUsed} gas`);
+            return txReceipt;
+        },
+
+        fastForwardToNextEpochAsync: async (): Promise<void> => {
+            // increase timestamp of next block
+            const epochDurationInSeconds = await this.stakingProxyContract.getEpochDurationInSeconds.callAsync();
+            await this._web3Wrapper.increaseTimeAsync(epochDurationInSeconds.toNumber());
+            // mine next block
+            await this._web3Wrapper.mineBlockAsync();
+        },
+
+        skipToNextEpochAsync: async (): Promise<TransactionReceiptWithDecodedLogs> => {
+            await this.utils.fastForwardToNextEpochAsync();
+            // increment epoch in contracts
+            const txReceipt = await this.utils.goToNextEpochAsync();
+            // mine next block
+            await this._web3Wrapper.mineBlockAsync();
+            return txReceipt;
+        },
+
+        // Other Utils
+        moveStakeAsync: async (
```

Comment:
If we create a constructor in `StakeInfo` that takes an optional `poolId` then we could also nix this function :)